presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotColumnUtils.java

```diff
@@ -39,14 +51,60 @@ private PinotColumnUtils()
     {
         return pinotTableSchema.getColumnNames().stream()
                 .filter(columnName -> !columnName.startsWith("$")) // Hidden columns starts with "$", ignore them as we can't use them in PQL
-                .map(columnName -> new PinotColumn(columnName, getPrestoTypeFromPinotType(pinotTableSchema.getFieldSpecFor(columnName))))
+                .map(columnName -> new PinotColumn(columnName, getPrestoTypeFromPinotType(pinotTableSchema.getFieldSpecFor(columnName)), isNullableColumnFromPinotType(pinotTableSchema.getFieldSpecFor(columnName)),
+                        getCommentFromPinotType(pinotTableSchema.getFieldSpecFor(columnName))))
                 .collect(toImmutableList());
     }
 
+    private static String getCommentFromPinotType(FieldSpec field)
+    {
+        return field.getFieldType().name();
+    }
+
+    private static boolean isNullableColumnFromPinotType(FieldSpec field)
+    {
+        return DEFAULT_PINOT_COLUMN_IS_NULLABLE;
+    }
+
     public static Type getPrestoTypeFromPinotType(FieldSpec field)
     {
         if (field.isSingleValueField()) {
-            return getPrestoTypeFromPinotType(field.getDataType());
+            switch (field.getFieldType()) {
+                case TIME:
+                    TimeFieldSpec timeFieldSpec = (TimeFieldSpec) field;
+                    TimeGranularitySpec outSpec = timeFieldSpec.getOutgoingGranularitySpec();
+                    if (outSpec != null) {
+                        if (outSpec.getTimeFormat().equalsIgnoreCase(TimeGranularitySpec.TimeFormat.EPOCH.name())) {
+                            if ((TimeUnit.DAYS == outSpec.getTimeType()) && (outSpec.getTimeUnitSize() == 1)) {
+                                return DateType.DATE;
+                            }
+                            if ((TimeUnit.MILLISECONDS == outSpec.getTimeType()) && (outSpec.getTimeUnitSize() == 1)) {
+                                return TimestampType.TIMESTAMP;
+                            }
+                        }
+                    }
+                    else {
+                        TimeGranularitySpec inSpec = timeFieldSpec.getIncomingGranularitySpec();
+                        if ((TimeUnit.DAYS == inSpec.getTimeType()) && (inSpec.getTimeUnitSize() == 1)) {
+                            return DateType.DATE;
+                        }
+                        if ((TimeUnit.MILLISECONDS == inSpec.getTimeType()) && (inSpec.getTimeUnitSize() == 1)) {
+                            return TimestampType.TIMESTAMP;
+                        }
+                    }
+                    return getPrestoTypeFromPinotType(field.getDataType());
```

Comment:
If we don't match the constraints above, should this be a bug ? Should we fail or should we continue to treat a time/date as an int ?

Reply:
yes, still treat it as int/long/string, as pinot support multiple timeformat like secondsSinceEpoch,hoursSinceEpoch, or just simpleDateFormat with custom udfs